src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,792 @@
+
+   if isdefined(x, :descr)
+      if x.descr==:GU || x.descr==:SU
+         print(io, string(x.descr), "(",x.deg,",",characteristic(x.ring)^(div(degree(x.ring),2)),")")
+      else
+         print(io, string(x.descr), "(",x.deg,",",order(x.ring),")")
+      end
+   else
+      print(io, "Matrix group of degree ", x.deg, " over ")
+      show(IOContext(io, :compact => true), x.ring)
+   end
+end
+
+function Base.show(io::IO, x::MatrixGroupElem)
+   if isdefined(x, :elm)
+      show(io, "text/plain", x.elm)
+#      print(io, x.elm)
+   else
+      print(io, GAP.gap_to_julia(GAP.Globals.StringViewObj(x.X)))
+   end
+end
+
+group_element(G::MatrixGroup, x::GapObj) = MatrixGroupElem(G,x)
+
+function assign_from_description(G::MatrixGroup)
+   if G.descr==:GL G.X=GAP.Globals.GL(G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==:SL G.X=GAP.Globals.SL(G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==:Sp G.X=GAP.Globals.Sp(G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==Symbol("GO+") G.X=GAP.Globals.GO(1,G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==Symbol("SO+") G.X=GAP.Globals.SO(1,G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==Symbol("Omega+") G.X=GAP.Globals.Omega(1,G.deg,Int(order(G.ring)))
+   elseif G.descr==Symbol("GO-") G.X=GAP.Globals.GO(-1,G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==Symbol("SO-") G.X=GAP.Globals.SO(-1,G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.Omega(-1,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GO G.X=GAP.Globals.GO(0,G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==:SO G.X=GAP.Globals.SO(0,G.deg,G.mat_iso.fr.codomain)
+   elseif G.descr==:Omega G.X=GAP.Globals.Omega(0,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GU G.X=GAP.Globals.GU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   elseif G.descr==:SU G.X=GAP.Globals.SU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   else error("unsupported description")
+   end
+end
+
+# return the G.sym if isdefined(G, :sym); otherwise, the field :sym is computed and set using information from other defined fields
+# NOTE: if G.X has to be set, then also the field G.mat_iso is set
+function Base.getproperty(G::MatrixGroup, sym::Symbol)
+
+   isdefined(G,sym) && return getfield(G,sym)
+
+   if sym === :mat_iso
+      G.mat_iso = gen_mat_iso(G.deg, G.ring)
+
+   elseif sym === :X
+      if isdefined(G,:descr)
+         if !isdefined(G,:mat_iso)
+            G.mat_iso = gen_mat_iso(G.deg, G.ring)
+         end
+         assign_from_description(G)
+      elseif isdefined(G,:gens)
+         if !isdefined(G,:mat_iso)
+            G.mat_iso = gen_mat_iso(G.deg, G.ring)
+         end
+         V = GAP.julia_to_gap([g.X for g in gens(G)])
+         G.X=GAP.Globals.Group(V)
+      else
+         error("Cannot determine underlying GAP object")
+      end
+   end
+
+   return getfield(G, sym)
+
+end
+
+
+function Base.getproperty(x::MatrixGroupElem, sym::Symbol)
+
+   if isdefined(x,sym) return getfield(x,sym) end
```

Comment:
For consistency with the preceding function...
```suggestion
   isdefined(x,sym) && return getfield(x,sym)
```